### PR TITLE
[WIP] Build with Swift Package Manager

### LINKED ISSUE #2: Build with the Swift Package Manager
This is the base requirement for working on Linux.

We'll need an LLVM and clang modulemap, and we'll need to be able to integrate the runtime target.

### PR DESCRIPTION
**Pull request by harlanhaskins**
This patch migrates Trill to a modular build system with a CMake driver
that invokes swift build with the appropriate flags to build the LLVM
wrappers. It also includes a build script so contributors don't need to
use CMake.

Closes #2.

### CODE REVIEW

File: Sources/AST/ASTDumper.swift [hunk 1]
```diff
@@ -87,243 +93,264 @@ class ASTDumper<StreamType: ColoredStream>: ASTTransformer {
...
-  override func visitCoercionExpr(_ expr: CoercionExpr) -> Void {
+  public override func visitCoercionExpr(_ expr: CoercionExpr) -> Void {
     printNode(expr) {
       super.visitCoercionExpr(expr)
     }
   }
 }
+
+extension String {
+    func splitCapitals() -> [String] {
```
Comment (segiddins) [inline Sources/AST/ASTDumper.swift:339]: this is very not-unicode aware. Couldn't you instead split by characters, and check to see if `c.uppercased == c`?

**Comment by harlanhaskins** [comment]
Okay so it builds but the build script needs more work to actually report rest failures as failures!

File: Sources/AST/WitnessTable.swift [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+/// Copyright 2016-2017 the Trill project authors.
+/// Licensed under the MIT License.
+///
+/// Full license text available at https://github.com/trill-lang/trill
+///
+
+import Source
+
+public struct WitnessTable {
+    public let proto: ProtocolDecl
```
Comment (segiddins) [inline Sources/AST/WitnessTable.swift:13]: indentation

File: README.md [hunk 3]
```diff
@@ -32,42 +27,33 @@ func main() {
...
-}
+Trill builds on macOS and Linux using CMake. We have a convenient build script that obviates the need for using CMake directly.
+Once you've got LLVM and CMake installed, you'll need to generate pkgconfig files for LLVM and Clang -- we have a tool for this in the
+`utils` directory. You'll only need to run it once.
 
-func main() {
-  println(Date().getTime())
-}
+```bash
+swift utils/make-pkgconfig.swift
```
Comment (segiddins) [inline README.md]: this seems to be gone?

File: README.md [hunk 4]
```diff
@@ -32,42 +27,33 @@ func main() {
...
 
-## Building and Using
-
-Trill currently is only supported on macOS, with the following dependencies:
+Then, you should be able to run our build script. Just running the build script gets you a build of `trill` in the `.build/debug` folder.
 
-- LLVMSwift and TinyGC, which are vendored as git submodules. Run `git
-  submodule update --init` to clone them.
+```bash
+usage: build-script [-h] [--swift SWIFT] [-r RELEASE] [-x] [-t]
```
Comment (segiddins) [inline README.md]: I'm not seeing this build script anywhere?

**Review by segiddins — COMMENTED** [review]
```
/Users/segiddins/Development/OpenSource/Trill/Sources/LLVMWrappers/src/LLVMWrappers.cpp:21:10: fatal error: 'clang-c/Platform.h' file not found
#include <clang-c/Platform.h>
         ^~~~~~~~~~~~~~~~~~~~
1 error generated.
```

File: README.md (hunk 4 above)
Comment (harlanhaskins) [reply README.md]: I need to rename that to `utils/build`

File: README.md (hunk 3 above)
Comment (harlanhaskins) [reply README.md]: I’m debating making that file...
I could just merge the make-pkgconfig files from LLVMSwift and ClangSwift and just put them in this repo?

File: Sources/AST/WitnessTable.swift (hunk 2 above)
Comment (harlanhaskins) [reply Sources/AST/WitnessTable.swift:13]: Thanks

**Comment by harlanhaskins** [comment]
Okay here's what I'm gonna do. I'm gonna merge this, then rebase the overloading PR on this. THEN I'm gonna fix the build errors as part of that PR to get all of this unstuck.

**Comment by segiddins** [comment]
💕

**Pull request merged**